### PR TITLE
Autoinstall frida devkits

### PR DESCRIPTION
**Pull request by s1341**
This PR implements automatic download and installation of the neccessary frida devkits.

At the moment, the download_and_uncompress_devkit is copied to both frida-gum-sys/build.rs and frida-sys/build.rs, so that they are independent. 

Tested and working.

### CODE REVIEW

**Comment by s1341** [comment]
Looks like the CI is failing, because it does the same thing... it downloads and installs the devkits. I recommend dropping that from the CI script.

**Comment by meme** [comment]
Thank you for the PR. I have a few issues with this change, though. Firstly, I am not a fan of downloading and installing arbitrary code without user consent during the build process. I would be OK, however, if we required the user to specify an environment variable during `cargo build` like `FRIDA_ENABLE_DOWNLOAD` which must be set. Then in the READMEs for the crates we can add this note, otherwise the user can install the devkit manually.

Secondly, the code duplication here is a bit unfortunate. Especially since we would now have compatible Frida version numbers in the READMEs and these build scripts which is tedious to update. If possible, see if you can pull the `download_and_uncompress_devkit` function out into a new crate called `frida-build` in the root (or something similar) and make it a build dependency of both crates. Then we can make a file in the root of the repository called `FRIDA_VERSION` which contains a string like `14.2.3` and is `include_bytes!`'d in the dependent crates. This way, when I make a release I just have to edit a single file.

Please let me know what you think and we can get this merged.

**Comment by s1341** [comment]
Ok.

I hear what you are saying about the download. Can I recommend that instead of an environment variable we use a feature flag? This will make it easier to include this crate in other packages?

About the code duplication. I totally agree: it stinks. I just wanted to get something working. I will create the frida-build crate as you recommend.

**Comment by s1341** [comment]
Implemented your suggestions. At the moment I've made the "autodownload" feature flag default. We can change that if you want.

Ready to merge from my perspective.

File: frida-sys/frida-core.h [hunk 1]
```diff
@@ -219,6 +219,7 @@
 #define g_assertion_message_cmpstrv _frida_g_assertion_message_cmpstrv
 #define g_assertion_message_error _frida_g_assertion_message_error
 #define g_assertion_message_expr _frida_g_assertion_message_expr
+#define g_assertion_set_handler _frida_g_assertion_set_handler
```
Comment (meme) [inline frida-sys/frida-core.h:222]: Please uncheck this file from this PR. I updated Frida recently and this appears to be the old one?

**Review by meme — CHANGES_REQUESTED** [review]
Looks good, but please make `autodownload` be `auto-download` and not the default behaviour. Then define the same feature in both `frida` and `frida-gum` so that downstream crates can enable it. (see the others for an example)

**Comment by s1341** [comment]
I'll get to your request tomorrow.

I have some other questions I'd like to ask. Are you available on IRC/discord/twitter somewhere we could chat a bit?

**Comment by meme** [comment]
Sure, `keegan` on `irc.gnome.org`.

**Comment by s1341** [comment]
Implemented your suggested changes.

**Comment by meme** [comment]
Thanks!

**Pull request merged**